packages/hardhat-core/test/internal/util/download.ts

```diff
@@ -51,7 +51,7 @@ describe("Compiler List download with proxy", function () {
     it("Should call download with the right params", async function () {
       const compilersDir = this.tmpDir;
       const downloadPath = path.join(compilersDir, "downloadedCompilerProxy");
-      const expectedUrl = `https://solc-bin.ethereum.org/wasm/list.json`;
+      const expectedUrl = `http://solc-bin.ethereum.org/wasm/list.json`;
```

Comment:
I suspect that this is hiding a bug. Note that this is testing an https proxy.

Reply:
@fvictorio you mentioned something about this.

Reply:
If I change it back to `https`, I get the error message `Error: Failed to download https://solc-bin.ethereum.org/wasm/list.json - 400 received. Only "http:" protocol prefix is supported`. That message appears to come from the npm package `proxy` that we're using for the test, which has: 
```
		if ('http:' != parsed.protocol) {
			// only "http://" is supported, "https://" should use CONNECT method
			res.writeHead(400);
			res.end('Only "http:" protocol prefix is supported\n');
			return;
		}
```
(https://github.com/TooTallNate/proxy/blob/d677ef31fd4ca9f7e868b34c18b9cb22b0ff69da/proxy.js#L213-L218)

Who is it that should be using `CONNECT`? I would guess that it should be `undici`, specifically its `ProxyAgent` that we're using.

Reply:
Let's defer this to a future PR. I created this issue: https://github.com/NomicFoundation/hardhat/issues/2418